Lib/shlex.py

```diff
@@ -22,6 +22,10 @@ def __init__(self, instream=None, infile=None, posix=False,
                  punctuation_chars=False):
         if isinstance(instream, str):
             instream = StringIO(instream)
+        elif isinstance(instream, bytes):
+            # convert byte instreams to string
+            instream = StringIO(instream.decode("ascii", "surrogateescape"))
```

Comment:
Why would you only support ASCII strings? for POSIX platforms, you should support also additional characters (see wordchars being extended).